source/Random.cpp

```diff
@@ -104,12 +109,53 @@ uint32_t Random::Binomial(uint32_t t, double p)
 
 
 
-// Get a normally distributed number (mean = 0, sigma= 1).
-double Random::Normal()
+// Get a normally distributed number (mean = 0, sigma= 1) using std::normal_distribution.
+double Random::StdNormal()
 {
 	normal_distribution<double> normal;
 #ifndef __linux__
 	lock_guard<mutex> lock(workaroundMutex);
 #endif
 	return normal(gen);
 }
+
+
+
+// Get a normally distributed number (mean = 0, sigma= 1) using std::normal_distribution.
+double Random::StdNormal(double mean, double sigma)
+{
+	normal_distribution<double> normal{mean, sigma};
+#ifndef __linux__
+	lock_guard<mutex> lock(workaroundMutex);
+#endif
+	return normal(gen);
+}
+
+
+
+// Get a normally distributed number (mean = 0, sigma = 1) using the Box-Muller transform.
+// Cache the unused value without transforming it so that it can be transformed when it's used.
+double Random::BMNormal(double mean, double sigma)
+{
+	if(normalBMCached)
+	{
+		normalBMCached = false;
+		return sigma * cachedBMNormal + mean;
```

Comment:
I don't really understand your explanation as to why I can't cache the value like this.  I'll try to address it in greater detail when I can.

As far as for all means and sigmas, I took care of that a couple days ago myself.  Again, the date discrepancy is confusing me.

Reply:
If two threads use this method, there is a data race. Both threads read a non-thread-local value not held by a lock and at least one will attempt to update it.

Reply:
So do you think this can be safely cached at all?

Reply:
What would your opinion be on using something from the C++17 standard (~4.5 years old at this point)?  I can't get it to work (build) yet but I'm hoping `[ThreadStatic]` will resolve this issue.  As I understand it this will force each thread to use its own instance of the cache variables meaning that while we won't be able to cache everything, we'll be able to cache the vast majority.

Reply:
Ok so I'm not *completely* hopeless: I noticed the thread_local usage on the mutex and distributions with the Random includes.  I'm just going to need some time to apply that to my code, I think.

Reply:
Is this okay: 2b3f65dc616c77c0c1df2b27f1c406e5a91c4fa4?  After investigating a little bit, I pretty much just copied what was already there.

Reply:
Alright this doesn't even work.  I mean it builds but it hangs in the changed code.

Reply:
It doesn't hang anymore: 9e65b65715b1b4ce73890088983a089cf3519e5d.  Hopefully it does address the race condition.